components/CompanyRegister.tsx

```diff
@@ -1,42 +1,72 @@
-import { CreateCompanyItem } from '@/db/entities/company';
+import { yupResolver } from '@hookform/resolvers/yup';
+import * as yup from 'yup';
+import { useForm } from 'react-hook-form';
+import { useAccount } from 'wagmi';
 import useCompany from '@/hooks/useCompanyRegistration';
-import { isSuccessResponse } from '@/pages/api/companies';
+import { useRecruitmentRegisterCompany } from '../src/generated';
 import { useRouter } from 'next/router';
 import { useEffect, useState } from 'react';
 import { toast } from 'react-toastify';
-import { useAccount } from 'wagmi';
-import { useRecruitmentRegisterCompany } from '../src/generated';
+import { isSuccessResponse } from '@/pages/api/companies';
 import { Badge } from './ui/badge';
 
+const schema = yup.object().shape({
+  companyName: yup.string().required('Company Name is required'),
+  companyEmail: yup
+    .string()
+    .email('Invalid email')
+    .required('Email is required'),
+  companySite: yup
+    .string()
+    .matches(
+      /^(https?:\/\/)?(www\.)?([A-Za-z0-9-._~:/?#[\]@!$&'()*+,;=]|%[A-Fa-f0-9]{2})*$/,
+      'Invalid website URL'
+    )
+  });
+
+interface FormData {
+  companyName: string;
+  companyEmail: string;
+  companySite: string;
+}
+
 const CompanyRegister = () => {
   const router = useRouter();
   const { address } = useAccount();
-  const { isValidating, isRegistered } = useCompany(address);
+  const { isRegistered } = useCompany(address);
   const { isLoading, isSuccess, writeAsync } = useRecruitmentRegisterCompany();
 
-  const [companyName, setCompanyName] = useState('');
-  const [companyEmail, setCompanyEmail] = useState('');
-  const [companySite, setCompanySite] = useState('');
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    getValues,
+  } = useForm<FormData>({
+    resolver: yupResolver(schema) as any,
```

Comment:
we want to use `any` as little as possible in our code

```suggestion
    resolver: yupResolver(schema),
```

Reply:
done